numpy/doc/structured_arrays.py

```diff
@@ -6,231 +6,506 @@
 Introduction
 ============
 
-NumPy provides powerful capabilities to create arrays of structured datatype.
-These arrays permit one to manipulate the data by named fields. A simple 
-example will show what is meant.: ::
+Structured arrays are ndarrays whose datatype is composed of simpler datatypes,
+which are organized as a sequence of named :term:`fields <field>`. For example, ::
 
- >>> x = np.array([(1,2.,'Hello'), (2,3.,"World")],
- ...              dtype=[('foo', 'i4'),('bar', 'f4'), ('baz', 'S10')])
+ >>> x = np.array([('Rex', 9, 81.0), ('Fido', 3, 27.0)],
+ ...              dtype=[('name', 'U10'), ('age', 'i4'), ('weight', 'f4')])
  >>> x
- array([(1, 2.0, 'Hello'), (2, 3.0, 'World')],
-      dtype=[('foo', '>i4'), ('bar', '>f4'), ('baz', '|S10')])
+ array([('Rex', 9, 81.0), ('Fido', 3, 27.0)],
+       dtype=[('name', 'S10'), ('age', '<i4'), ('weight', '<f4')])
 
-Here we have created a one-dimensional array of length 2. Each element of
-this array is a structure that contains three items, a 32-bit integer, a 32-bit
-float, and a string of length 10 or less. If we index this array at the second
-position we get the second structure: ::
+Here ``x`` is a one-dimensional array length 2, whose datatype is a structure
```

Comment:
"array of length two" and omit the comma. The clause after the comma is essential.